### DISCUSSION TITLE
Can integrations involve a paid service?

### DISCUSSION THREAD

**Discussion by adamhenson** [vercel/community#2322] [Help] [answered]
### Summary

Hello - I own [foo.software](https://www.foo.software) which provides free and premium (paid) tiers of service so users can monitor their website's page experience (Lighthouse scores with insights). I'd like to create a Vercel integration, but it would need to only be accessible to Foo's registered users (and would probably need to have tiers in the Vercel integration... to limit number of URLs / runs).

Before I invest a ton of time in creating this integration, I wanted to first find out if this is even allowed by Vercel.

These are the last 2 steps in which a user connects to an integration per the [docs](https://vercel.com/docs/integrations/create-integration#redirect-url). I've added a difference that includes details of what would occur if the user was not yet registered with Foo.

```diff
 - Allow users to provision a new or connect existing projects in your system with their Vercel Projects
+   - User either proceeds with login or registration flow of integration app. If registering - the user may choose a premium plan which would require adding a payment method.
 - Upon completion, redirect the user back to Vercel.
```

## Questions

1. Would the above be "allowed" by Vercel?
2. From what I can see, the marketplace doesn't support paid integrations (billed by Vercel), which would be ideal here. I know this is a new feature, so any plans for this?

**Comment by amyegan** [comment, marked as answer]
Hi @adamhenson! I don't believe Vercel has support for payment collection on behalf of an integration provider. But you are still welcome to offer different tiers on your service. If you're interested, I recommend visiting our [Technology Partners](https://vercel.com/partners/technology) page and reaching out to the partnerships team from there.

**Comment by adamhenson** [comment]
Thanks @amyegan
